packages/blade/src/components/Dropdown/_decisions/decisions.md

```diff
@@ -0,0 +1,703 @@
+  - [ActionListSection](#actionlistsection)
+  - [ActionListHeader](#actionlistheader)
+  - [ActionListFooter](#actionlistfooter)
+- [Why `x` instead of `y`](#why-x-instead-of-y)
+- [Accessibility](#accessibility)
+- [Open Questions](#open-questions)
+- [Referrences](#referrences)
+
+---
+
+<br/>
+
+## Dropdown
+
+`Dropdown` will be a wrapper component which will handle the logic of opening and closing of the dropdown. It won't separately render anything in UI apart from the first children. (E.g. `SelectInput` in above example)
+
+### API
+
+Usage
+
+```jsx
+<Dropdown>{/* Dropdown sub-components */}</Dropdown>
+```
+
+Props
+
+_No Props Needed_
+
+### SelectInput
+
+<img src="2022-11-24-16-33-30.png" alt="Select Field Figma Designs" width="500" />
+
+Select is an Input Field that can open a dropdown on click. It can be extended from `BaseInput` component which covers most decisions like validations, focus, etc.
+
+#### Designs
+
+- [Figma - Select](https://www.figma.com/file/jubmQL9Z8V7881ayUD95ps/Blade---Payment-Light?node-id=13590%3A171090&t=KTWGvEUBEuUdQh9K-0)
+
+#### API
+
+Sample Usage
+
+```jsx
+<SelectInput icon={CalendarIcon} label="Your favorite Design System" helpText="Select only one" />
+```
+
+Props
+
+Mostly similar to other `Input` components extended from `BaseInput`
+
+```ts
+type SelectInputProps = {
+  // ... FormLabelProps like `label`, `labelPosition`
+  // ... FormValidationProps like `validationState`, `helpText`, `errorText`, etc.
+  // ... State Props like `isRequired`, `isDisabled`,
+  icon: IconComponent;
+};
+```
+
+#### A11y
+
+Would be ideal if we can render it as `button` with `aria-expanded` prop to handle state of dropdown open and close.
+
+Reference: https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-expanded#buttons
+
+### `DropdownOverlay`
+
+An Overlay that positions dropdown correctly.
+
+Should handle the animations for dropdown.
+
+#### API
+
+Usage
+
+```jsx
+<Dropdown>
+  <SelectInput label="Click For Greetings" />
+  <DropdownOverlay>Hi</DropdownOverlay>
+</Dropdown>
+```
+
+Props
+
+_No Props Needed_
+
+<br/><br/>
+
+## ActionList
+
+Plain list that can be rendered inside dropdown, in navigation menu, or anywhere. It has selectable items that perform some action on click. It can also have multiple selectable items that render as checkboxes
+
+![](2022-12-08-16-58-46.png)
+
+### Design
+
+- [Figma - ActionList]()
+
+### API
+
+#### Simplest Usage Possible
+
+```jsx
+<ActionList
+  surfaceLevel={2}
+  onChange={(e) => {
+    console.log({ name: e.name, values: e.values });
+  }}
+>
+  <ActionListItem
+    title="Accept API"
+    value="accept"
+    description="Accept this API"
+    leading={<ActionListItemIcon icon={StarIcon} />}
+    trailing={<ActionListItemText>⌘ + Space + A</ActionListItemText>}
+    onClick={() => {}}
+  />
+  <ActionListItem
+    title="Delete"
+    description="Delete this value"
+    value="delete"
+    intent="negative"
```

Comment:
Can we have an `ActionListItem` with `intent="negative"` when ActionListProps.selectionType is `multiple`? If yes, we need to check with design as this would be a bad practice.

Reply:
confirmed from designer that it we don't have usecase for negative intent in multi select. I have updated the doc